ha-core-infra/src/main/java/org/kie/remote/impl/producer/EventProducer.java

```diff
@@ -60,6 +60,7 @@ protected void internalProduceSync(String topicName, String key, Object object)
             producer.send(getFreshProducerRecord(topicName, key, object)).get();
         } catch (InterruptedException | ExecutionException e) {
             logger.error("Error in produceSync!", e);
+            Thread.currentThread().interrupt();
```

Comment:
Can you please describe what this means. Wouldn't we rather restart the whole app at this point?

Reply:
I agree with Jiri, I think here we should update global status to make k8s aware of the failure and restart the pod

Reply:
@jiripetrlik @danielezonca  
The Thread interrupt was asked by Sonarcloud,
three bugs are on the LeaderImplemetation copied from Camel, Sonarcloud is asking to split methods because are too complex, 
three smell are on same classes copied by camel,
are we sure to change classes copied from other libraries without test in their original project ?
the fourth code smell is to add a test to a util method Test...

Reply:
The .get() called on the Future of the producer ask to catch a InterruptedException and Execution exception, then Sonarcloud ask to add a call to interrupt on the current thread, I'm happy to remove but this increase the sonarcloud warnings

Reply:
I don't mean to remove that interrupt. Sonar cloud is probably right if it asks for something like this. The problem is that if this happens something is probably wrong with app on the pod and it probably needs to be restarted. Am I right?

Reply:
Yep, you are right, let me add this change of Global status